openid-4-verifiable-presentations-1_0.md

```diff
@@ -616,11 +624,325 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
 
 If the Verifier responds with any HTTP error response, the Wallet MUST terminate the process.
 
+# Digital Credentials Query Language (DCQL) {#dcql_query}
+
+The Digital Credentials Query Language (DCQL, pronounced [ˈdakl̩]) is a
+JSON-encoded query language that allows the Verifier to request Verifiable
+Presentations that match the query. The Verifier MAY encode constraints on the
+combinations of credentials and claims that are requested. The Wallet evaluates
+the query against the Verifiable Credentials it holds and returns Verifiable
+Presentations matching the query.
+
+A valid DCQL query is defined as a JSON-encoded object with the following
+top-level properties:
+
+`credentials`:
+: REQUIRED. A non-empty array of Credential Queries as defined in (#credential_query)
+that specify the requested Verifiable Credentials.
+
+`credential_sets`:
+: OPTIONAL. A non-empty array of credential set queries as defined in (#credential_set_query)
+that specifies additional constraints on which of the requested Verifiable Credentials to return.
+
+Note: Future extensions may define additional properties both on the top level
+and in the rest of the DCQL data structure. Implementations MUST ignore any
+unknown properties.
+
+## Credential Query {#credential_query}
+
+A Credential Query is an object representing a request for a presentation of one
+Credential.
+
+Each entry in `credentials` MUST be an object with the following properties:
+
+`id`:
+: REQUIRED. A string identifying the Credential in the response and, if provided,
+the constraints in `credential_sets`. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the Authorization Request, the same `id` MUST NOT
+be present more than once.
+
+`format`:
+: REQUIRED. A string that specifies the format of the requested
+Verifiable Credential. Valid Credential Format Identifier values are defined in
+Appendix A of [@!OpenID.VCI].
+
+`meta`: 
+: OPTIONAL. An object defining additional properties requested by the Verifier that
+apply to the metadata and validity data of the Credential. The properties of
+this object are defined per Credential Format in (#format_specific_properties). If omitted,
+no specific constraints are placed on the metadata or validity of the requested
+Credential.
+
+`claims`:
+: OPTIONAL. A non-empty array of objects as defined in (#claims_query) that specifies
+claims in the requested Credentials.
```

Comment:
Maybe I'm not getting it but every other thing at this level is about a single credential so this one being about "requested Credentials" seems not right? 

```suggestion
`claims`:
: OPTIONAL. A non-empty array of objects as defined in (#claims_query) that specifies
claims in the requested Credential.
```